src/mango/src/mango_cursor_view.erl

```diff
@@ -482,6 +564,32 @@ apply_opts([{_, _} | Rest], Args) ->
     % Ignore unknown options
     apply_opts(Rest, Args).
 
+-spec consider_index_coverage(#idx{}, fields(), #mrargs{}) -> #mrargs{}.
+consider_index_coverage(Index, Fields, #mrargs{include_docs = IncludeDocs} = Args) ->
```

Comment:
I had noticed that we always calls `apply_opts` then right after always `call consider_index_coverage`. It might make sense to have a function like `apply_cursor_opts` which then may still call `apply_opts` and `consider_index_coverage` (since both are tested separately they can stay that way to minimize the changes diff).